Model/Sync/Catalog/Data.php

```diff
@@ -374,43 +383,42 @@ public function attributeMapping(Product $item)
 
         foreach ($mapAttributes as $key => $attr) {
             try {
-                if ($key === 'gtins') {
-                    $value = $this->prepareGtinsData($attr, $item);
-                } elseif ($key === 'custom_properties') {
-                    $value = $this->prepareCustomProperties($attr, $item);
-                } elseif ($key === 'is_discontinued') {
-                    $value = false;
-                } else {
-                    if ($attr['default']) {
-                        $data = $item->getData($attr['attr_code']);
-
-                        if (isset($attr['type']) && $attr['type'] === 'url') {
-                            $data = $item->getProductUrl();
+                switch ($key) {
+                    case 'gtins':
+                        $value = $this->prepareGtinsData($attr, $item);
+                        break;
+                    case 'custom_properties':
+                        $value = $this->prepareCustomProperties($attr, $item);
+                        break;
+                    case 'is_discontinued':
+                        $value = false;
+                        break;
+                    case 'url':
+                        $value = $item->getProductUrl();
+                        break;
+                    case 'image_url':
+                        $value = $this->getProductImageUrl($item);
+                        break;
+                    default:
+                        if (!$attr['default'] && isset($attr['method']) && $attr['method']) {
```

Comment:
Can we rename `attr` to `attributeMap`? And the `mappingAttributes` property to `attributesMapping`, and use it in the switch case instead of declaring it in line 382?